backend/tracim_backend/command/search.py

```diff
@@ -0,0 +1,111 @@
+            help="do not insered content after creating index template",
+            dest="empty",
+            required=False,
+            default=None,
+        )
+        return parser
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        # TODO - G.M - 05-04-2018 -Refactor this in order
+        # to not setup object var outside of __init__ .
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+        self.search_api = SearchFactory.get_search_lib(
+            current_user=None, session=self._session, config=self._app_config
+        )
+        self.search_api.create_index()
+        print("Index template was created")
+        if not parsed_args.empty:
+            print("Indexing all content")
+            self.search_api.index_all_content()
+            print("All content where indexed")
+
+
+class SearchIndexUpdateCommand(AppContextCommand):
+    def get_description(self) -> str:
+        return "update index of search engine"
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        # TODO - G.M - 05-04-2018 -Refactor this in order
+        # to not setup object var outside of __init__ .
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+        self.search_api = SearchFactory.get_search_lib(
+            current_user=None, session=self._session, config=self._app_config
+        )
+        self.search_api.migrate_index()
+
+
+class SearchIndexAddCommand(AppContextCommand):
+    def get_description(self) -> str:
+        return "Add content into search engine"
+
+    def get_parser(self, prog_name: str) -> argparse.ArgumentParser:
+        parser = super().get_parser(prog_name)
+        parser.add_argument(
+            "--content_id",
+            help="select a specific content_id to index, "
+            "if not provided will index all content of tracim",
+            dest="content_id",
+            required=False,
+            default=None,
+        )
+        return parser
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        # TODO - G.M - 05-04-2018 -Refactor this in order
+        # to not setup object var outside of __init__ .
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+        self.search_api = SearchFactory.get_search_lib(
+            current_user=None, session=self._session, config=self._app_config
+        )
+        if parsed_args.content_id:
+            content_api = ContentApi(
+                current_user=None, session=self._session, config=self._app_config
+            )
+            content = content_api.get_one(
+                content_id=int(parsed_args.content_id), content_type=content_type_list.Any_SLUG
```

Comment:
For info: instead make `int(parsed_args.content_id)` use `type=int` parameter of `parser.add_argument`.

Reply:
done

Reply:
doen